plugins/cairo-lang-macro/src/types/token.rs

```diff
@@ -0,0 +1,130 @@
+use std::fmt::Display;
+
+use cairo_lang_filesystem::span::{TextOffset, TextSpan, TextWidth};
+use cairo_lang_parser::utils::SimpleParserDatabase;
+use cairo_lang_syntax::node::{db::SyntaxGroup, SyntaxNode};
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+
+/// Representation of most atomic struct that is passed within host<->macro communication.
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct Token {
+    span: TextSpan,
+    content: String,
+}
+
+impl Token {
+    pub fn new(content: String, span: TextSpan) -> Self {
+        Self { content, span }
+    }
+
+    pub fn from_syntax_node(db: &dyn SyntaxGroup, node: SyntaxNode) -> Self {
+        Token::new(node.get_text(db), node.span(db))
+    }
+}
+
+/// An abstract stream of Cairo tokens.
+///
+/// This is both input and part of an output of a procedural macro.
+#[derive(Debug, Default, Clone)]
+pub struct TokenStream {
+    pub tokens: Vec<Token>,
+    pub metadata: TokenStreamMetadata,
+}
+
+/// Metadata of [`TokenStream`].
+///
+/// This struct can be used to describe the origin of the [`TokenStream`].
+#[derive(Debug, Default, Clone)]
+pub struct TokenStreamMetadata {
+    /// The path to the file from which the [`TokenStream`] has been created.
+    pub original_file_path: Option<String>,
+    /// ID of the file from which the [`TokenStream`] has been created.
+    ///
+    /// It is guaranteed, that the `file_id` will be unique for each file.
+    pub file_id: Option<String>,
+}
+
+impl TokenStream {
+    #[doc(hidden)]
+    pub fn new(tokens: Vec<Token>) -> Self {
+        Self {
+            tokens,
+            metadata: TokenStreamMetadata::default(),
+        }
+    }
+
+    pub fn from_string(str: String) -> Self {
+        let db = SimpleParserDatabase::default();
+        // Sometimes this will return an error, when the user pass the simple literal value like "34".
```

Comment:
This will fail if there is passed expression instead of `ModuleItem`, when parsing virtual file it's kind have to be correctly set